docs/guides/how_to_guides/creating_and_editing_expectations/how_to_contribute_a_new_expectation_to_great_expectations.rst

```diff
@@ -0,0 +1,411 @@
+
+
+            * Implement the computation of the Metric in your new Metric Provider class for at least one of the three backends (Execution Engines) that Great Expectations supports: pandas, sqlalchemy, spark. Most contributors find starting with Pandas is the easiest and fastest way to build.
+
+                The parent class of your Metric Provider class is ``ColumnMapMetricProvider``. It uses Python Decorators to hide most of the complexity from you and give you a clear and simple API to implement one method per backend that computes the metric.
+
+                .. admonition:: Note:
+
+                    - If you have never used Python Decorators and don't know what they are and how they work, no worries - this should not stop you from successfully implementing your Expectation. Decorators allow the parent class to "wrap" your methods, which means to execute some code before and after your method runs. All you need to know is the name of the Decorator to add (with "@") above your method definition.
+
+                Find the following code snippet in your Metric Provider class:
+
+                .. code-block:: python
+
+                    @column_condition_partial(engine=PandasExecutionPandasExecutionEngineEngine)
+                    def _pandas(cls, column, **kwargs):
+                        return column == 3
+
+
+
+                This means that the method ``_pandas`` is a metric function that is decorated as a ``column_condition_partial``. It will be called with the engine-specific column type (e.g., a Series in pandas case). It must return a boolean value for each row of the column.
+                The ``engine`` argument of ``column_condition_partial`` is set to ``PandasExecutionEngine`` to signal to the method in the parent that the method computes the Metric for pandas backend.
+                There is nothing special about the name of the method ``_pandas`` - it can be called anything else, but why make things more complicated than they must be?
+
+                Implement this method to compute your Metric.
+
+                .. admonition:: Note:
+
+                    How to support additional arguments your Expectation needs.
+
+                    The Expectation in the template (``expect_column_values_to_equal_three``) did not need to accept any additional arguments to evaluate the data.
+
+                    Here is how you could modify ``expect_column_values_to_equal_three`` to ``expect_column_values_to_equal_integer``, where users would have to specify the value of the integer as an argument:
+
+                    * Find the snippet ``success_keys = ("mostly",)`` in the class that implements your Expectation. Add your arguments to ``success_keys``
+
+                    .. code-block:: python
+
+                        success_keys = ("mostly", "integer")
```

Comment:
These are ordered (sort of ... in practice users just need to be explicit in naming kwargs), but I think integer should come first to maintain the usual signature where mostly is at the end.

Reply:
+1